server/src/test/java/org/elasticsearch/common/util/ArrayUtilsTests.java

```diff
@@ -30,15 +30,15 @@
 public class ArrayUtilsTests extends ESTestCase {
     public void testBinarySearch() throws Exception {
         for (int j = 0; j < 100; j++) {
-            int index = Math.min(randomInt(0, 10), 9);
-            double tolerance = Math.random() * 0.01;
-            double lookForValue = randomFreq(0.9) ? -1 : Double.NaN; // sometimes we'll look for NaN
+            int index = randomIntBetween(0, 9);
+            double tolerance = randomDoubleBetween(0, 0.01, true);
+            double lookForValue = frequently() ? -1 : Double.NaN; // sometimes we'll look for NaN
             double[] array = new double[10];
             for (int i = 0; i < array.length; i++) {
                 double value;
-                if (randomFreq(0.9)) {
-                    value = Math.random() * 10;
-                    array[i] = value + ((randomFreq(0.5) ? 1 : -1) * Math.random() * tolerance);
+                if (frequently()) {
+                    value = randomDoubleBetween(0, 9, true);
```

Comment:
wasn't the test generating doubles between 0 and 10 before? (it might not matter)

Reply:
Math.random() -> "a pseudorandom {@code double} greater than or equal to {@code 0.0} and less than {@code 1.0}", so I think 10 wasn't included. But yes, I agree it probably doesn't matter at all ;-)